ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/OperationsReOrgManager.java

```diff
@@ -0,0 +1,143 @@
+        recentChainData.getEveryRootOnChainTillSlot(commonAncestorSlot, oldBestBlockRoot);
+    notCanonicalBlockRoots.forEach(
+        (__, root) -> {
+          SafeFuture<Optional<BeaconBlock>> maybeBlockFuture =
+              recentChainData.retrieveBlockByRoot(root);
+          maybeBlockFuture
+              .thenAccept(
+                  maybeBlock ->
+                      maybeBlock.ifPresentOrElse(
+                          block -> {
+                            BeaconBlockBody blockBody = block.getBody();
+                            proposerSlashingPool.addAll(blockBody.getProposer_slashings());
+                            attesterSlashingPool.addAll(blockBody.getAttester_slashings());
+                            exitPool.addAll(blockBody.getVoluntary_exits());
+
+                            // Attestations need to get re-processed through AttestationManager
+                            // because we don't have access to the state with which they were
+                            // verified anymore and we need to make sure later on
+                            // that they're being included on the correct fork.
+                            blockBody
+                                .getAttestations()
+                                .forEach(
+                                    attestation -> {
+                                      attestationManager
+                                          .onAttestation(attestation)
+                                          .finish(
+                                              result ->
+                                                  result.ifInvalid(
+                                                      reason ->
+                                                          LOG.debug(
+                                                              "Rejected re-queued attestation from block: {} due to: {}",
+                                                              root,
+                                                              reason)),
+                                              err ->
+                                                  LOG.error(
+                                                      "Failed to process re-queued attestation from block: {} due to: {}",
+                                                      root,
+                                                      err));
+                                    });
+                          },
+                          () ->
+                              LOG.debug(
+                                  "Failed to re-queue operations for now non-canonical block: {}",
+                                  root)))
+              .finish(
+                  err ->
+                      LOG.warn(
+                          "Failed to re-queue operations for now non-canonical block: {} due to future error: {}",
+                          root,
+                          err.getMessage()));
```

Comment:
We should preserve the stack trace instead of only logging the error message.
```suggestion
                      LOG.warn(
                          "Failed to re-queue operations for now non-canonical block: {}",
                          root,
                          err));
```

Reply:
Done.